app/src/main/java/org/oppia/android/app/player/audio/AudioViewModel.kt

```diff
@@ -66,9 +66,11 @@ class AudioViewModel @Inject constructor(
     processPlayStatusLiveData()
   }
 
-  fun setStateAndExplorationId(newState: State, id: String) {
-    state = newState
-    explorationId = id
+  fun setStateAndExplorationId(newState: State?, id: String?) {
+    if (newState != null && id != null) {
+      state = newState
+      explorationId = id
+    }
```

Comment:
I have traced the usage of this function to: https://github.com/oppia/oppia-android/blob/f7e46519f54ef37871cefce27de1812bf4510493/app/src/main/java/org/oppia/android/app/player/state/StatePlayerRecyclerViewAssembler.kt#L276-L283. Here, I can see that we perform a nullcheck for `id`, but not for `state`. As state is the variable throwing the uninitialized property exception, it might be worth finding a more indpeth reason why it is not being initialized on time. Or perharbs both values are not being initialized on time.

Reply:
Hey @adhiamboperes, 
Thanks for the insight! I will also ensure to add null checks for both state and id to prevent the issue. 
I'll keep you updated on my progress!